tools/link-validation/link_validator.py

```
@@ -61,13 +87,41 @@ def _build_file_cache(self) -> None:
+        settings = self.config.get("validation_settings", {})
+        
+        # Skip DNS prefetch and similar rel attributes
+        rel_attr = link_attributes.get("rel", "").lower()
+        skip_rel_attrs = exclusions.get("skip_rel_attributes", [])
+        if rel_attr in skip_rel_attrs:
+            return True, f"Skipped: rel='{rel_attr}' link (DNS prefetch/preconnect)"
+        
+        # Skip specific DNS prefetch links
+        dns_prefetch_links = exclusions.get("dns_prefetch_links", [])
+        if link in dns_prefetch_links:
+            return True, f"Skipped: Known DNS prefetch link"
```

Ruff (0.12.2)

f-string without any placeholders

Remove extraneous `f` prefix

(F541)